src/Electrical/Digital/logic_vectors.jl

```diff
@@ -0,0 +1,47 @@
+import Base: size, axes, getindex
+
+const LogicOrNumber = Union{Logic, Number}
+
+struct StdULogicVector{N} <: AbstractArray{Logic, N}
+    logic::AbstractArray{Logic}
+    level::AbstractArray{Int}
+    function StdULogicVector(l::AbstractArray)
+        N = ndims(l)
+        l = AbstractArray{Logic}(convert.(Logic, l))
+        new{N}(l, get_logic_level.(l))
+    end
+end
+
+struct StdLogicVector{N} <: AbstractArray{Logic, N}
+    logic::AbstractArray{Logic}
+    level::AbstractArray{Int}
+    function StdLogicVector(l::AbstractArray)
+        N = ndims(l)
+        l = AbstractArray{Logic}(convert.(Logic, l))
+        new{N}(l, get_logic_level.(l))
+    end
+end
+
+const LogicVector = Union{StdULogicVector, StdLogicVector}
+
+Base.size(l::L) where {L <: LogicVector} = size(l.logic)
```

Comment:
```suggestion
Base.size(l::LogicVector) = size(l.logic)
```
this is against the Julia style guide ;)

Reply:
There are several similar instances below. These give precompilation warnings and are best avoided